### DISCUSSION TITLE
rbenv install 2.7.2 BUILD failed

### DISCUSSION THREAD

**Discussion by kaymo4** [rbenv/ruby-build#1864] [Build failures]
I checked rbenv is installed ok:
curl -fsSL https://github.com/rbenv/rbenv-installer/raw/master/bin/rbenv-doctor | bash
Checking for `rbenv' in PATH: /usr/local/bin/rbenv
Checking for rbenv shims in PATH: OK
Checking `rbenv install' support: /usr/local/bin/rbenv-install (ruby-build 20210405)
Counting installed Ruby versions: none
  There aren't any Ruby versions installed under `/Users/myname/.rbenv/versions'.
  You can install Ruby versions like so: rbenv install 2.2.4
Checking RubyGems settings: OK
Auditing installed plugins: OK

Here is what happens:
==================
% rbenv install 2.7.2
Installing openssl-1.1.1j...
Installed openssl-1.1.1j to /Users/myname/.rbenv/versions/2.7.2

Downloading ruby-2.7.2.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.7/ruby-2.7.2.tar.bz2
Installing ruby-2.7.2...
ruby-build: using readline from homebrew

**BUILD FAILED (macOS 11.2.3 using ruby-build 20210405)**

Inspect or clean up the working tree at /var/folders/t3/29rsxx6s4pn3x21y1x3nqdpc0000gn/T/ruby-build.20210407161251.50329.vjt7f4

**Comment by maisaroq** [comment]
Having the same issue on macOS 11.2.3 on Apple Silicon.

**Comment by FiorellaChilcon** [comment]
same here, I have this error
`BUILD FAILED (macOS 11.2.3 using ruby-build 20210405)`

**Comment by lobo-az** [comment]
I'm the same.

```
BUILD FAILED (macOS 11.2.3 using ruby-build 20210405)

Inspect or clean up the working tree at /var/folders/5f/l_7000d55_l4f6jsb18j63hm0000gn/T/ruby-build.20210414212243.66826.mXczth
Results logged to /var/folders/5f/l_7000d55_l4f6jsb18j63hm0000gn/T/ruby-build.20210414212243.66826.log

Last 10 log lines:
        from ./tool/rbinstall.rb:846:in `block (2 levels) in install_default_gem'
        from ./tool/rbinstall.rb:279:in `open_for_install'
        from ./tool/rbinstall.rb:845:in `block in install_default_gem'
        from ./tool/rbinstall.rb:835:in `each'
        from ./tool/rbinstall.rb:835:in `install_default_gem'
        from ./tool/rbinstall.rb:799:in `block in <main>'
        from ./tool/rbinstall.rb:950:in `block in <main>'
        from ./tool/rbinstall.rb:947:in `each'
        from ./tool/rbinstall.rb:947:in `<main>'
make: *** [do-install-all] Error 1
```

**Comment by garveychan** [comment]
I had no luck with reinstalling command line tools, configuring openssl, reformatting, etc. but **pausing my anti-virus** got it working for me. Hopefully saves someone else a few hours of head scratching.

**Comment by jasonwells** [comment]
> pausing my anti-virus got it working for me

@garveycodes Thanks! Turning off anti-virus was the solution for one of our engineers as well. Sohpos is what they were running.

**Comment by MatthewSteen** [comment]
> 
> 
> > pausing my anti-virus got it working for me
> 
> @garveycodes Thanks! Turning off anti-virus was the solution for one of our engineers as well. Sohpos is what they were running.

Suspending my VPN allowed me to install 2.7.2.

**Comment by tomholford** [comment]
This resolved the issue for Linux:

```
sudo apt-get install libssl-dev libreadline-dev
```

Source:
https://stackoverflow.com/a/64729884
